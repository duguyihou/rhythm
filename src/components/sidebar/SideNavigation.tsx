import React from 'react'

import {
  faCirclePlay,
  faBorderAll,
  IconDefinition,
} from '@fortawesome/free-solid-svg-icons'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import clsx from 'clsx'
import Link from 'next/link'

type SideNavigationItem = {
  name: string
  to: string
  icon: IconDefinition
}

const SideNavigation = () => {
  const navigation: SideNavigationItem[] = [
    { name: 'Listen Now', to: '/', icon: faCirclePlay },
    { name: 'Browse', to: '/browse', icon: faBorderAll },
  ]
  return (
    <>
      {navigation.map(({ name, to, icon }) => (
        <Link key={name} href={to}>
          <div
            className={clsx(
              'text-gray-700 hover:bg-gray-300 hover:text-gray-900 cursor-pointer',
              'group flex items-center px-2 py-2 text-base font-medium rounded-md'
            )}
          >
            <div className="w-4 h-4">
              <FontAwesomeIcon icon={icon} />
            </div>
            <div className="px-2">{name}</div>
          </div>
        </Link>
      ))}
    </>
  )
}

export default SideNavigation
