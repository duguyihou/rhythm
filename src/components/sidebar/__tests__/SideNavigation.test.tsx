import { render } from '@testing-library/react'

import SideNavigation from '../SideNavigation'

test('sideNavigation', () => {
  const sideNavigation = render(<SideNavigation />)

  expect(sideNavigation).toMatchSnapshot()
})
