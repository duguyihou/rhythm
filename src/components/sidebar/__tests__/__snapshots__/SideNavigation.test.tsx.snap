// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sideNavigation 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="text-gray-700 hover:bg-gray-300 hover:text-gray-900 cursor-pointer group flex items-center px-2 py-2 text-base font-medium rounded-md"
      >
        <div
          class="w-4 h-4"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-circle-play "
            data-icon="circle-play"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M512 256c0 141.4-114.6 256-256 256S0 397.4 0 256S114.6 0 256 0S512 114.6 512 256zM188.3 147.1c-7.6 4.2-12.3 12.3-12.3 20.9V344c0 8.7 4.7 16.7 12.3 20.9s16.8 4.1 24.3-.5l144-88c7.1-4.4 11.5-12.1 11.5-20.5s-4.4-16.1-11.5-20.5l-144-88c-7.4-4.5-16.7-4.7-24.3-.5z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="px-2"
        >
          Listen Now
        </div>
      </div>
      <div
        class="text-gray-700 hover:bg-gray-300 hover:text-gray-900 cursor-pointer group flex items-center px-2 py-2 text-base font-medium rounded-md"
      >
        <div
          class="w-4 h-4"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-border-all "
            data-icon="border-all"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M384 96V224H256V96H384zm0 192V416H256V288H384zM192 224H64V96H192V224zM64 288H192V416H64V288zM64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H384c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64z"
              fill="currentColor"
            />
          </svg>
        </div>
        <div
          class="px-2"
        >
          Browse
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="text-gray-700 hover:bg-gray-300 hover:text-gray-900 cursor-pointer group flex items-center px-2 py-2 text-base font-medium rounded-md"
    >
      <div
        class="w-4 h-4"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-circle-play "
          data-icon="circle-play"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M512 256c0 141.4-114.6 256-256 256S0 397.4 0 256S114.6 0 256 0S512 114.6 512 256zM188.3 147.1c-7.6 4.2-12.3 12.3-12.3 20.9V344c0 8.7 4.7 16.7 12.3 20.9s16.8 4.1 24.3-.5l144-88c7.1-4.4 11.5-12.1 11.5-20.5s-4.4-16.1-11.5-20.5l-144-88c-7.4-4.5-16.7-4.7-24.3-.5z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="px-2"
      >
        Listen Now
      </div>
    </div>
    <div
      class="text-gray-700 hover:bg-gray-300 hover:text-gray-900 cursor-pointer group flex items-center px-2 py-2 text-base font-medium rounded-md"
    >
      <div
        class="w-4 h-4"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-border-all "
          data-icon="border-all"
          data-prefix="fas"
          focusable="false"
          role="img"
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M384 96V224H256V96H384zm0 192V416H256V288H384zM192 224H64V96H192V224zM64 288H192V416H64V288zM64 32C28.7 32 0 60.7 0 96V416c0 35.3 28.7 64 64 64H384c35.3 0 64-28.7 64-64V96c0-35.3-28.7-64-64-64H64z"
            fill="currentColor"
          />
        </svg>
      </div>
      <div
        class="px-2"
      >
        Browse
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
